rustup-init.sh

```diff
@@ -540,7 +540,7 @@ get_architecture() {
     # and fall back to arm.
     # See https://github.com/rust-lang/rustup.rs/issues/587.
     if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
-        if ensure grep '^Features' /proc/cpuinfo | grep -E -q -v 'neon|simd'; then
```

Comment:
Currently, if `grep` fails (because e.g. /proc/cpuinfo is unavailable or malformed), we still assume `armv7`. This doesn't work with the case described in https://github.com/rust-lang/rustup/issues/2700#issue-841531263 when we're using ARMv6.